src/Repository/ScheduleRepository.php

```diff
@@ -71,8 +71,8 @@ public function regenerateScheduleEvents($schedule) : Collection
         $parser_dates = $parser->getDates();
         $current_dates = $schedule->scheduleEvents()->get()->pluck('date');
         $deprecated_dates = collect($current_dates)->diff($parser_dates)->toArray();
-        $new_dates = collect($parser_dates)->diff($current_dates)->filter(function($date) use ($today) {
-            return $date >= $today;
+        $new_dates = collect($parser_dates)->diff($current_dates)->filter(function($date) use ($today, $from_today) {
```

Comment:
If you need an external conditional (e.g. `$from_today`) you can use `when` as shown below:

```php
$new_dates = collect($parser_dates)->diff($current_dates)->when($from_today, function($date) use ($today) {
    return $date >= $today;
});
```

Reply:
Nice. I'll tweak that

Reply:
Ah, not on 5.3 though